android/src/main/java/com/thebluealliance/androidclient/activities/LaunchActivity.java

```diff
@@ -92,227 +66,65 @@ protected void onCreate(Bundle savedInstanceState) {
                 goToHome();
                 return;
             }
+        } else if (redownload) {
+            // Start redownload activity
+            startActivity(redownloadIntent);
+        } else {
+            // Go to onboarding activity
+            startActivity(new Intent(this, OnboardingActivity.class));
+            finish();
         }
-        setContentView(R.layout.activity_launch);
-        viewPager = (DisableSwipeViewPager) findViewById(R.id.view_pager);
-        viewPager.setSwipeEnabled(false);
-        viewPager.setOffscreenPageLimit(2);
-        viewPager.setAdapter(new FirstLaunchFragmentAdapter(this));
-        loadingMessage = (TextView) findViewById(R.id.message);
-
-        // If the activity is being recreated after a config change, restore the message that was
-        // being shown when the last activity was destroyed
-        if (savedInstanceState != null) {
-            if (savedInstanceState.containsKey(CURRENT_LOADING_MESSAGE_KEY)) {
-                currentLoadingMessage = savedInstanceState.getString(CURRENT_LOADING_MESSAGE_KEY);
-                loadingMessage.setText(currentLoadingMessage);
-            }
-        }
-        findViewById(R.id.welcome_next_page).setOnClickListener(this);
-        findViewById(R.id.finish).setOnClickListener(this);
-        if (redownload) {
-            ((TextView) findViewById(R.id.welcome_message)).setText(getString(R.string.update_message));
-        }
-        loadFragment = (LoadAllDataTaskFragment) getFragmentManager().findFragmentByTag(LOAD_FRAGMENT_TAG);
-        if (loadFragment != null) {
-            viewPager.setCurrentItem(1, false);
-        }
-    }
-
-    @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        super.onSaveInstanceState(outState);
-        outState.putString(CURRENT_LOADING_MESSAGE_KEY, currentLoadingMessage);
     }
 
-    private boolean checkDataRedownload() {
+    private boolean checkDataRedownload(Intent intent) {
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
```

Comment:
While you're at this, the `SharedPreference` instance should come from DI (the MyTbaModule)